internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -177,6 +194,14 @@ func (connector *SingleUseConnector) Connect(
 		loginContext,
 		ctxtypes.ClientLoginKey,
 		clientLoginChannel)
+	loginContext = context.WithValue(
+		loginContext,
+		ctxtypes.ServerLoginAckKey,
+		loginAckChannel)
+	loginContext = context.WithValue(
+		loginContext,
+		ctxtypes.ServerErrorKey,
+		errChannel)
```

Comment:
Similar to how we created a `ConnectResult` to treat "the returned value and err" as a single unit, we should create a struct to hold all of these different channel types.  Then we'll only need to do a single `WithValue` here.  And we'll have a single type to represent "all our communication with the driver" rather than having 4 different scattered channels.